lua/damagelogs/shared/autoslay.lua

```diff
@@ -373,30 +375,21 @@ if CLIENT then
             table.insert(walls, net.ReadEntity())
         end
 
-        for _, v in pairs(walls) do
-            table.insert(ents, v)
+        for _, v in ipairs(walls) do
+            table.insert(jails, v)
         end
     end)
 
-    hook.Add("Think", "JailWalls", function()
-        local function CheckWalls()
-            local found = false
-
-            for k, v in pairs(ents) do
-                if IsValid(v) then
-                    v:SetCustomCollisionCheck(true)
-                    v.jailWall = true
-                    table.remove(ents, k)
-                    found = true
-                    break
-                end
-            end
-
-            if found then
-                CheckWalls()
+    local function CheckWalls()
+        for k, v in ipairs(jails) do
+            if IsValid(v) then
+                v:SetCustomCollisionCheck(true)
+                v.jailWall = true
+                table.remove(jails, k)
```

Comment:
Sorry for the delay.

I believe the reason why this function called itself recursively was to get around the issue caused by removing items from a table while looping over the table.

Here's a simpler example
```lua
local test = {}
test[1] = "hello"
test[2] = "world"
test[3] = "nice"

for k, v in ipairs(test) do
  if(k == 2) then
    table.remove(test, k)
  else
    print(k.." "..v)
  end
end
```

This results in the output
```
1 hello
```


This code can be massively simplified to this:
```lua
local function CheckWalls()
    for k, v in ipairs(jails) do
        if IsValid(v) then
            v:SetCustomCollisionCheck(true)
            v.jailWall = true
        end
    end

    jails = {}
end
```

Reply:
agreed, fixed.